docs/developer/release-cycle.md

```diff
@@ -0,0 +1,69 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`).
```

Comment:
nit:
```suggestion
The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`) is generated.
```

Reply:
I'll use "is released". I think generated is not the right word